### PR TITLE
refactor: Added the difference between working of finetuning vs transfer learning

### PR DESCRIPTION
**Pull request by Anindyadeep**
Since the finetuning page has concepts for both images and text. For texts fine-tuning term is commonly used but for images, sometimes people confuses the differences between the two.

# Review checklist

> Don't worry about satisfying all items, it's fine to open a (draft) PR.

- [ ] chapter content
  + [ ] only one top-level `# h1-Title`
  + [ ] summary (e.g. table or TL;DR overview), no need for an explicit `## Summary/Introduction` title or equivalent
  + [ ] main content focus: recent developments in open source AI
    + general context/background (brief)
    + current pros/cons
    + in-depth insights (not yet widely known)
  + [ ] likely `## Future` developments
  + [ ] end with `{{ comments }}`
- [ ] appropriate citations
  + [ ] BibTeX references
  + [ ] Glossary terms
  + [ ] cross-references (figures/chapters)
  + [ ] (if `new-chapter.md`), add `_toc.yml` entry & `index.md` table row
  + [ ] If CI URL checks have false-positives, append to `_config.yml:sphinx.config.linkcheck*`
- [ ] images & data not committed to this repo (e.g. use https://github.com/premAI-io/static.premai.io instead)

### CODE REVIEW

**Comment by Anindyadeep** [comment]
The first commit is only for addressing the difference between the two. I will either add more commit or may be a different PR for Parameter efficient Finetuning methods.

File: fine-tuning.md [hunk 1]
```diff
@@ -19,17 +19,46 @@ Some ideas:
...
+2. Freeze the weights of the pre-trained layers so they remain fixed during training. This retains all the original knowledge. However sometimes you can also un-freeze some of the pretrained layers (i.e. last set of pretrained layers) for further tuning.
+3. Add a new layer / head on the top of the forzen model. This layer is been added to capture more task/domain specific patterns.
+4. Train on your new downstream dataset by passing batches of data through the model architecture and comparing outputs to true labels.
+5. Continue training the model until the task layer and selected pre-trained layers converge on optimal weights for your dataset.
+
+The key is that here the original weights of the pre-trained model remains unchanged and will not be trained (hence the term freezed). We train the new set of layers appeneded on the top of the pretrain model to get trained for a specific task. The pretrain model just acts as a generalized, robust feature extractor. One thing to note here is: Transfer learning can be applied on if the dataset (the one which we will be used in transfer learning) size is descent.
```
Comment (casperdcl) [inline fine-tuning.md:35]: are you sure?

**Review by casperdcl — CHANGES_REQUESTED** [review]
> Transfer learning can be applied on if the dataset (the one which we will be used in transfer learning) size is [large].

and

> Finetuning approach is often used when the dataset (that will be used for finetuning) is scarce.

are you sure it's not the other way around? transfer learning = mostly frozen, few unfrozen layers to train on minimal data, right? Meanwhile full fine-tuning is like retraining from scratch with a "good" weight initialisation, lots of unfrozen (trainable) weights, lots of data needed?

File: fine-tuning.md [hunk 2]
```diff
@@ -19,17 +19,46 @@ Some ideas:
...
+2. Now, depending on the architecture and the size of the model, we can either train all the layers of the model (full-finetuning) or freeze some parts of the model and keep the rest trainable (we can also term this scenerio as fine-tuning).
+3. Also for tasks like classification, a general practice includes removing the classifier layer of the pretrained model and adding an untrained classifier layer head.
+4. Train on your new downstream dataset by passing batches of data through the model architecture and comparing outputs to true labels.
+6. Continue training the model until the selected pre-trained layers (or sometimes the full model) converge on optimal weights for your dataset.
+
+The key is that most of the original model weights remain fixed during training. Only a small portion of weights are updated to customise the model to new data. This transfers general knowledge while adapting for the new/domain task at the same time.
+
+Finetuning approach is often used when the dataset (that will be used for finetuning) is scarce. And in that case, we need to train parts of the model (or sometimes the whole model) to adapt on the task, trained using the small dataset.
```
Comment (casperdcl) [inline fine-tuning.md:53]: are you sure?

**Comment by Anindyadeep** [comment]
Hi @casperdcl, I am closing this PR. Because I am starting to contribute through my forked repo from now onwards.

**Pull request closed without merging**